src/platform_impl/linux/x11/window.rs

```diff
@@ -1719,16 +1719,17 @@ impl UnownedWindow {
 
     #[inline]
     pub fn request_user_attention(&self, request_type: Option<UserAttentionType>) {
-        let mut wm_hints =
+        if let Some(mut wm_hints) =
             WmHints::get(self.xconn.xcb_connection(), self.xwindow as xproto::Window)
-                .expect("Failed to get WM hints")
-                .reply()
-                .expect("Failed to get WM hints reply");
-        wm_hints.urgent = request_type.is_some();
-        wm_hints
-            .set(self.xconn.xcb_connection(), self.xwindow as xproto::Window)
-            .expect("Failed to set WM hints")
-            .ignore_error();
+                .ok()
+                .and_then(|cookie| cookie.reply().ok())
+        {
```

Comment:
Doesn't this just mean that this function doesn't do anything anymore?

Since things previously failed due to a missing WM hints property, I guess that nothing sets that property by default. So, this function also won't.

Instead of `if let Some(mut wm_hints) = ....`, I suggest to do `let mut wm_hints = ....or_default()`. That matches more closely what the old C code does (assuming I found the correct piece of code and assuming that `alloc_wm_hints` just calls `XAllocWMHints` which zero-initializes the structure):
```rust
impl XConnection {
    pub fn get_wm_hints(
        &self,
        window: ffi::Window,
    ) -> Result<XSmartPointer<'_, ffi::XWMHints>, XError> {
        let wm_hints = unsafe { (self.xlib.XGetWMHints)(self.display, window) };
        self.check_errors()?;
        let wm_hints = if wm_hints.is_null() {
            self.alloc_wm_hints()
        } else {
            XSmartPointer::new(self, wm_hints).unwrap()
        };
        Ok(wm_hints)
    }
```

Reply:
That sounds like a plan